examples/mortality.py

```diff
@@ -0,0 +1,276 @@
+from numpyro.infer.reparam import LocScaleReparam
+
+
+def get_data():
+    # NEED TO USE from numpyro.examples.datasets import MORTALITY, load_dataset
+    data = pd.read_csv("simulated_mortality.csv")
+    print(data.head())
+
+    grid_lookup = data[["s1", "s2"]].drop_duplicates().sort_values(by="s2")
+
+    return (
+        data["a"].values,
+        data["s2"].values,
+        data["t"].values,
+        grid_lookup["s1"].values,
+        data["population"].values,
+        data["deaths"].values,
+    )
+
+
+reparam_config = {
+    k: LocScaleReparam(0)
+    for k in [
+        "alpha_s1",
+        "beta_s1",
+        "alpha_s2",
+        "beta_s2",
+        "alpha_age_drift",
+        "beta_age_drift",
+        "xi",
+        "nu_drift",
+        "gamma_drift",
+    ]
+}
+
+
+@numpyro.handlers.reparam(config=reparam_config)
+def model(age, space, time, lookup, population, deaths=None):
+    N_s1 = len(np.unique(lookup))
+    N_s2 = len(np.unique(space))
+    N_age = len(np.unique(age))
+    N_t = len(np.unique(time))
+    N = len(population)
+
+    # plates
+    age_plate = numpyro.plate("age_groups", N_age, dim=-3)
+    space_plate = numpyro.plate("space", N_s2, dim=-2)
+    year_plate = numpyro.plate("year", N_t - 1, dim=-1)
+
+    # hyperparameters
+    sigma_alpha_s1 = numpyro.sample("sigma_alpha_s1", dist.Uniform(0.0, 2.0))
+    sigma_beta_s1 = numpyro.sample("sigma_beta_s1", dist.Uniform(0.0, 2.0))
+    sigma_alpha_s2 = numpyro.sample("sigma_alpha_s2", dist.Uniform(0.0, 2.0))
+    sigma_beta_s2 = numpyro.sample("sigma_beta_s2", dist.Uniform(0.0, 2.0))
+    sigma_alpha_age = numpyro.sample("sigma_alpha_age", dist.Uniform(0.0, 2.0))
+    sigma_beta_age = numpyro.sample("sigma_beta_age", dist.Uniform(0.0, 2.0))
+    sigma_xi = numpyro.sample("sigma_xi", dist.Uniform(0.0, 2.0))
+    sigma_nu = numpyro.sample("sigma_nu", dist.Uniform(0.0, 2.0))
+    sigma_gamma = numpyro.sample("sigma_gamma", dist.Uniform(0.0, 2.0))
+    theta = numpyro.sample("theta", dist.Exponential(0.1))
+
+    # spatial hierarchy
+    with numpyro.plate("s1", N_s1, dim=-2):
+        alpha_s1 = numpyro.sample("alpha_s1", dist.Normal(0, sigma_alpha_s1))
+        beta_s1 = numpyro.sample("beta_s1", dist.Normal(0, sigma_beta_s1))
+    with space_plate:
+        alpha_s2 = numpyro.sample(
+            "alpha_s2", dist.Normal(alpha_s1[lookup], sigma_alpha_s2)
+        )
+        beta_s2 = numpyro.sample("beta_s2", dist.Normal(beta_s1[lookup], sigma_beta_s2))
+
+    # age
+    with age_plate:
+        alpha_age_drift_scale = jnp.pad(
+            jnp.broadcast_to(sigma_alpha_age, N_age - 1),
+            (1, 0),
+            constant_values=100.0,  # pad so first term is alpha0, the global intercept with prior N(0, 100)
+        )[:, jnp.newaxis, jnp.newaxis]
+        alpha_age_drift = numpyro.sample(
+            "alpha_age_drift", dist.Normal(0, alpha_age_drift_scale)
+        )
+        alpha_age = jnp.cumsum(alpha_age_drift, -2)
```

Comment:
I guess we need to cummulative sum over axis=-3 for `alpha_age` and `beta_age`?

Reply:
Very good spot. It's interesting that the model still ran with axis=-2. I guess the shape of `alpha_age` is `(19, 1, 1)` so summing over `-2` wouldn't sum anything. And in the case of the incorrect code above, `alpha_age_drift` and `alpha_age` are the same. So the code still ran but it wasn't correct.